doc/secciones/04_planificacion.tex

```diff
@@ -121,7 +122,7 @@ \section{Control de la calidad}
 
 \begin{itemize}
 	\item{Errores, adherencia a la guía de estilo \href{https://peps.python.org/pep-0008/}{PEP8} y complejidad del código.}
-	\item{Test unitarios, de integración y de cobertura}
+	\item{Test unitarios, de integración y de cobertura. Los tests han de definirse cubriendo un único caso lógico por cada test, lo que permite saber con precisión que funciona exactamente y que no, y cubriendo la mayor cantidad posible de casos lógicos. Para aceptar un PMV al menos el 85\% de su código debe estar cubierto por tests.}
```

Comment:
REcuerda hacer los tests de integración e identificarlos como tales.
En el caso de una aplicación CLI, tendrías que hacerlos desde un runner externo.

Reply:
Estaba entendiendo test de integración como integración de código entre librerías incluidas librerías de terceros. Ahora mismo se ejecuta la aplicación entera simulando que le llegan distintos argumentos cambiando sys.argv, solo se mockea el leer archivos como tal para usar los datos del test.

Poner un runner externo supongo que requiere configurar el runner, un CI aparte y en vez de usar los datos de prueba definidos en los tests poner ficheros de ejemplo en el repo.

No lo veo muy limpio pudiendo hacerlo dentro de pytest, casi que prefiero quitar que se hacen test de integración de la memoria si la terminología no es correcta.

Reply:
Sí, pero también estarían tets end-to-end. En todo caso para el CLI era solo un ejemplo, pero sí es cierto que tendrías que comprobar que va bien, está bien instalado y demás, y que los paths están correctos, los permisos correctos. Sería integración con el sistema operativo.
Técnicamente, la integración con librerías externas tiene que testearse, claro, pero desde el punto de vista del usuario final los end to end tienen más importancia.
Teniendo en cuenta, en todo caso, que hay aproximadamente un 0% de TFGs que los hacen, sólo si lo consideras necesario y si te sobra tiempo.

Reply:
Vale, entiendo la importancia e incluso imagino que sería útil probar en varios sistemas operativos. No creo que me de tiempo pero añado un issue porque ciertamente estaría bien tener tests end-to-end.